flytekit/control_plane/nodes.py

```diff
@@ -279,18 +275,53 @@ def promote_from_model(cls, base_model: _node_execution_models.NodeExecution) ->
             closure=base_model.closure, id=base_model.id, input_uri=base_model.input_uri, metadata=base_model.metadata
         )
 
+    def get_interface(self) -> "flytekit.control_plane.interface.TypedInterface":
+        """
+        Return the interface of the task or subworkflow associated with this node execution.
+        """
+        from flytekit.control_plane.tasks.task import FlyteTask
+        from flytekit.control_plane.workflow import FlyteWorkflow
+
+        if not self.metadata.is_parent_node:
+            # if not a parent node, assume a task execution node
+            task_id = self.task_executions[0].id.task_id
+            task = FlyteTask.fetch(task_id.project, task_id.domain, task_id.name, task_id.version)
+            return task.interface
+
+        # otherwise assume the node is associated with a subworkflow
+        client = _flyte_engine.get_client()
+        # need to get the FlyteWorkflow associated with this node execution (self), so we need to fetch the parent
```

Comment:
@katrogan added a comment here to explain this weirdness... can you think of an easier/cleaner way of doing this?

Reply:
oh this is quite annoying! can we cache the interface so we don't have to make successive rpcs?